docs/DeveloperGuide.md

```diff
@@ -1,37 +1,209 @@
+
+
+
+#### Sequence Diagram
+
+The following sequence diagram illustrates how `Parser` parse command for `AddTaskCommand`:
+
+![Parse_Add_Task_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/ParseAddTaskClassDiagram.png)
+
+A closer look on how `AddTaskCommand` is executed is shown below:
+
+![Add_Task_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/AddTaskClassDiagram.png)
+
+#### Implementation considerations:
+The `AddTaskCommand` has been implemented with a design that prioritizes scalability and maintainability. This allows the system to easily support additional task types in the future without requiring major code changes. For example, if new task types are needed in the future (e.g., `event`, `appointment`), they can be added by simply extending the Task class hierarchy and updating the createTask() factory method.
+### **FindCommand**
+
+The find feature allows users to find the name of a patient, or the name of a task. This is facilitated by the `FindPatientCommand` and `FindTaskCommand`.
+
+1. **User Input**: The user enters the `find` command followed by keywords
+2. **Command Parsing**: The `Parser` parses the input and creates a `FindParser` object.
+3. **Execution**: The `FindPatientCommand` or `FindTaskCommand` searches for the task.
+
+#### Sequence Diagram
+
+The following sequence diagram illustrates how the `FindPatientCommand` is executed:
+
+![Find_Sequence_Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/FindSequenceDiagram.png)
+
+### **State Switching Feature**
+
+The state switching feature allows the system to change its mode of operation between `MAIN_STATE` (for patient management) and `TASK_STATE` (for task management). This allows the software to handle commands differently based on the current state.
+
+1. **User Input**: The user enters a command pertaining either to patients (in `MAIN_STATE`) or tasks (in `TASK_STATE`).
+2. **Command Parsing**: The `Parser` checks the current state of the software by interacting with the `State` class. Depending on the state, it interprets  the command differently.
+3. **State Checking**: The `State` class tracks the current state of the application. If the system is in `MAIN_STATE`, the `Parser` creates patient-related commands (e.g., `AddPatientCommand`). If it's in `TASK_STATE`, task-related commands (e.g., `AddTaskCommand`)
+4. **Command Execution**: The appropriate command is then executed. For example, in `MAIN_STATE`, it adds a new patient, while in `TASK_STATE`, it adds a new task to a patient's task list.
+5. **State Transitions**: Commands like `SelectPatientCommand` and `BackCommand`, trigger state transitions.
+
+#### Sequence Diagram
+
+The following sequence diagram illustrates how the state-switching mechanism works:
+
+![State-Switching Sequence Diagram](https://github.com/AY2425S1-CS2113-T11-1/tp/raw/master/docs/images/StateSwitchingSequenceDiagram.png)
```

Comment:
I notice that some of your sequence diagrams use activation bars, but some of them not. You may consider to use a consistent way to structure you sequence diagrams.